rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,420 @@
+    private BtcBlockStore btcBlockStore;
+    private BridgeEventLogger bridgeEventLogger;
+    private FeePerKbSupport feePerKbSupport;
+    private Block initialBlock;
+    private StorageAccessor bridgeStorageAccessor;
+    private FederationStorageProvider federationStorageProvider;
+    private FederationSupportImpl federationSupport;
+    private LockingCapSupport lockingCapSupport;
+    private BridgeSupport bridgeSupport;
+
+    @Test
+    void whenAllActivationsArePresentAndFederationChanges_shouldCreateCommitAndActiveNewFed() throws Exception {
+        // Arrange
+        var activations = ActivationConfigsForTest.all().forBlock(0);
+        setUpFederationChange(activations);
+        // Create a default original federation using the list of UTXOs
+        var originalFederation = createOriginalFederation(
+            FederationType.P2SH_ERP, ORIGINAL_FEDERATION_KEYS, activations);
+        var originalUTXOs = federationStorageProvider.getNewFederationBtcUTXOs(
+            BRIDGE_CONSTANTS.getBtcParams(), activations);
+       
+        // Act
+        // Commit pending federation using the new federation
+        var newFederation = commitPendingFederation(NEW_FEDERATION_KEYS, activations);
+        // Since Lovell is activated we will commit the proposed federation
+        commitProposedFederation(activations);
+        // Move blockchain until the activation phase
+        activateNewFederation(activations);
+
+        // Assert
+        assertUTXOsMovedFromNewToOldFederation(originalUTXOs, activations);
+        assertNewFederationIsActive(newFederation.getAddress(), originalFederation.getAddress());
+        assertNoMigrationHasStarted();
+    }
+  
+    /* Change federation related methods */
+
+    private void setUpFederationChange(ActivationConfig.ForBlock activations) throws Exception {
+        repository = new MutableRepository(
+            new MutableTrieCache(
+                new MutableTrieImpl(null, new Trie())));
+        repository.addBalance(
+            PrecompiledContracts.BRIDGE_ADDR, co.rsk.core.Coin.fromBitcoin(BRIDGE_CONSTANTS.getMaxRbtc()));
+
+        bridgeStorageProvider = new BridgeStorageProvider(
+            repository,
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_CONSTANTS.getBtcParams(),
+            activations);
+
+        btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(
+            BRIDGE_CONSTANTS.getBtcParams(),
+            100,
+            100);
+        btcBlockStore = btcBlockStoreFactory.newInstance(
+            repository,
+            BRIDGE_CONSTANTS,
+            bridgeStorageProvider,
+            activations);
+        // Setting a chain head different from genesis to avoid having to read the checkpoints file
+        addNewBtcBlockOnTipOfChain(btcBlockStore);
+        repository.save();
+
+        bridgeEventLogger = new BridgeEventLoggerImpl(BRIDGE_CONSTANTS, activations, new ArrayList<>());
```

Comment:
```suggestion
        bridgeEventLogger = new BridgeEventLoggerImpl(BRIDGE_CONSTANTS, activations, logs);
```

we probably wanna define the logs at class level, to assert federation change related logged events. Wdyt?

Reply:
Sure, if in a future PR we need will make it an instance variable.

Reply:
wouldn't be worth to check for the commit federation event for example?